StripeCore/StripeCore/Source/Helpers/Timeout.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Timeout.swift
+//  StripeCore
+//
+//  Created by Joyce Qin on 10/28/25.
+//
+
+import Foundation
+
+@_spi(STP) public struct TimeoutError: Error {}
+
+// MARK: - Timeout with TaskGroup and CheckedContinuation escape hatch
+/// Runs multiple operations in parallel with individual timeouts
+/// - Parameters:
+///   - timeout: The maximum time interval to wait for each operation to complete
+///   - operations: Variadic operations to run with timeout
+/// - Returns: Tuple of Results, where each Result contains either the task's value or its error
```

Comment:
really important to call out that this returns a tuple of results _in the same order as the operations_.